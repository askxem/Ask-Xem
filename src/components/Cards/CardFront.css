figure {
    padding: 10px;
    margin: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: 1px solid gray;
    border-radius: 10%;
}