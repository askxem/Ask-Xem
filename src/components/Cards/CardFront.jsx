import styles from './CardFront.css' 

export default function CardFront({ card }) {
  return (
    <figure>
      <h3>{card.title}</h3>
      <img
          src={`/icons/${card.animal}.png`}
          alt={card.animal}
          className={styles.image}
      />
    </figure>
  )
}