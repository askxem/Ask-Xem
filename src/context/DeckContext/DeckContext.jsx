import { createContext, useState, useContext } from "react";

const DeckContext = createContext()

const DeckProvider = ({ children }) => {
    const [genSeen, setGenSeen] = useState(JSON.parse(localStorage.getItem('genSeen')) || [])
    const [pronSeen, setPronSeen] = useState(JSON.parse(localStorage.getItem('pronSeen')) || [])

    const seen = (id, category) => {

        if(category == 'gender') {
            let genSet = new Set(genSeen)
            genSet.add(id)
            const arr = Array.from(genSet)
            setGenSeen(arr)
            localStorage.setItem('genSEEN', JSON.stringify(arr))
        }
        if(category == 'pronoun') {
            let pronSet = new Set(pronSeen)
            pronSet.add(id)
            const arr = Array.from(pronSet)
            setPronSeen(arr)
            localStorage.setItem('pronSEEN', JSON.stringify(arr))
        }
    }

    return(
        <DeckContext.Provider value={{ genSeen, pronSeen, seen }}>{children}</DeckContext.Provider>
    )

}

const useDeck = () => {
    const context = useContext(DeckContext)
    if (context === undefined) {
        throw new Error('useDeck not used inside provider')
    }
    return context
}

export { DeckProvider, useDeck }