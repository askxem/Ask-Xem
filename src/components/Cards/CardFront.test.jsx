import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router'
import { DeckProvider } from '../../context/DeckContext/DeckContext'
import CardFront from './CardFront'

const character = {
  title: 'Sylvestor the Cat',
  animal: 'cat',
  image: 'testURL',
}
it('renders front card component', () => {
  const { container } = render(
    <DeckProvider>
      <MemoryRouter>
        <CardFront card={character} />
      </MemoryRouter>
    </DeckProvider>
  )

  expect(container).toMatchSnapshot()
})
