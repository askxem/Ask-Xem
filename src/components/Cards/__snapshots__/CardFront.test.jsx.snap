// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders front card component 1`] = `
<div>
  <figure
    class="frontcontainer"
  >
    <h3>
      Sylvestor the Cat
    </h3>
    <img
      alt="cat"
      class="image"
      src="/icons/cat.png"
    />
  </figure>
</div>
`;
